AirCasting/Graph/StatisticsContainer/Adapters/GraphStatsDataSource.swift

```diff
@@ -0,0 +1,27 @@
+// Created by Lunar on 06/07/2021.
+//
+
+import Foundation
+
+class GraphStatsDataSource: MeasurementsStatisticsDataSource {
+    let stream: MeasurementStreamEntity
+    
+    var dateRange: ClosedRange<Date> = Date.distantPast...Date.distantFuture
+    
+    init(stream: MeasurementStreamEntity) {
+        self.stream = stream
+    }
+    
+    var allMeasurements: [MeasurementStatistics.Measurement] {
```

Comment:
Well, my thinking was that it's in the struct which is created new every time we refresh the view. I did some testing and found two things. First is that the delegateRelay is indeed created new every time we refresh any of the properties (threshold, locationEnabled or pathpoints) but the new delegate isn't used anywhere. 
And the second thing is that I found a memory leak. It might be helpful for you to know that ;D 

![Screenshot 2021-08-06 at 09 09 30](https://user-images.githubusercontent.com/46443841/128471475-bf226781-973d-4e8b-9ba8-3922c1c6e2ae.png)